llama_index/evaluation/multi_modal/faithfulness.py

```diff
@@ -0,0 +1,222 @@
+
+    This evaluator only considers the response string and the list of context strings.
+
+    Args:
+        multi_modal_llm(Optional[MultiModalLLM]):
+            The Multi-Modal LLM Judge to use for evaluations.
+        raise_error(bool): Whether to raise an error when the response is invalid.
+            Defaults to False.
+        eval_template(Optional[Union[str, BasePromptTemplate]]):
+            The template to use for evaluation.
+        refine_template(Optional[Union[str, BasePromptTemplate]]):
+            The template to use for refining the evaluation.
+    """
+
+    def __init__(
+        self,
+        multi_modal_llm: MultiModalLLM | None = None,
+        raise_error: bool = False,
+        eval_template: str | BasePromptTemplate | None = None,
+        refine_template: str | BasePromptTemplate | None = None,
+    ) -> None:
+        """Init params."""
+        self._multi_modal_llm = multi_modal_llm or OpenAIMultiModal(
+            model="gpt-4-vision-preview", max_new_tokens=1000
+        )
+        self._raise_error = raise_error
+
+        self._eval_template: BasePromptTemplate
+        if isinstance(eval_template, str):
+            self._eval_template = PromptTemplate(eval_template)
+        else:
+            self._eval_template = eval_template or DEFAULT_EVAL_TEMPLATE
+
+        self._refine_template: BasePromptTemplate
+        if isinstance(refine_template, str):
+            self._refine_template = PromptTemplate(refine_template)
+        else:
+            self._refine_template = refine_template or DEFAULT_REFINE_TEMPLATE
+
+    def _get_prompts(self) -> PromptDictType:
+        """Get prompts."""
+        return {
+            "eval_template": self._eval_template,
+            "refine_template": self._refine_template,
+        }
+
+    def _update_prompts(self, prompts: PromptDictType) -> None:
+        """Update prompts."""
+        if "eval_template" in prompts:
+            self._eval_template = prompts["eval_template"]
+        if "refine_template" in prompts:
+            self._refine_template = prompts["refine_template"]
+
+    def evaluate(
+        self,
+        query: str | None = None,
+        response: str | None = None,
+        contexts: Sequence[str] | None = None,
+        image_paths: List[str] | None = None,
+        image_urls: List[str] | None = None,
+        **kwargs: Any,
+    ) -> EvaluationResult:
+        """Evaluate whether the response is faithful to the multi-modal contexts."""
+        del query  # Unused
+        del kwargs  # Unused
+        if contexts is None or response is None:
+            raise ValueError("contexts and response must be provided")
+
+        context_str = "\n\n".join(contexts)
+        fmt_prompt = self._eval_template.format(
+            context_str=context_str, query_str=response
+        )
+
+        image_documents = []
+        if image_paths:
+            for image_path in image_paths:
```

Comment:
If we have the paths, no need for the directory reader, we can directly create the image documents

`image_documents = [ImageDocument(image_path=image_path) for image_path in image_paths]`

Reply:
Actually, we could just directly create `ImageNode`s too, and not even need the splitter

Reply:
Yes, I'd love to go directly to `ImageNode` if possible.